jme3-desktop/src/main/java/com/jme3/system/NativeLibraryLoader.java

```diff
@@ -571,88 +478,69 @@ public static void loadNativeLibrary(String name, boolean isRequired) {
             loadedAsFileName = library.getExtractedAsName();
         } else {
             // Just use the original filename as it is in the JAR.
-            loadedAsFileName = fileNameInJar;
+            loadedAsFileName = Paths.get(pathInJar).getFileName().toString();
         }
         
         File extractionDirectory = getExtractionFolder();
         URLConnection conn;
-        InputStream in;
-        
+
         try {
             conn = url.openConnection();
-            in = conn.getInputStream();
         } catch (IOException ex) {
             // Maybe put more detail here? Not sure.
             throw new UncheckedIOException("Failed to open file: '" + url + 
                                            "'. Error: " + ex, ex);
         }
         
         File targetFile = new File(extractionDirectory, loadedAsFileName);
-        OutputStream out = null;
-        try {
+        try (InputStream in = conn.getInputStream()) {
+            boolean extract = true;
             if (targetFile.exists()) {
                 // OK, compare last modified date of this file to 
                 // file in jar
                 long targetLastModified = targetFile.lastModified();
                 long sourceLastModified = conn.getLastModified();
 
                 // Allow ~1 second range for OSes that only support low precision
-                if (targetLastModified + 1000 > sourceLastModified) {
-                    logger.log(Level.FINE, "Not copying library {0}. " + 
+                if (Math.abs(sourceLastModified - targetLastModified) < 1000) {

```

Comment:
This could be separated into a small private static function that returns a boolean indicating whether an operation should be executed, for example: 
```java
/**
* Tests whether 2 files are identical by comparing their last-date of modifications metadata, the last date of modification should be 1 second or more for the source and target to be considered as non-identical.
*
* @param source the source file
* @param target the target file 
* @return true if the difference in ldm is less than 1 second, false otherwise
*/
private static boolean hasIdenticalLdm(final URLConnection source, final File target) { 
     return Math.abs(source.getLastModified() - target.lastModified()) < 1000;
}
```
Where; `Ldm` stands for Last-Date-Of-Modification.

Then, remove `extract = false` and use the newly helper method in the proceeding checks.

Reply:
You may also change the `hasIdenticalLdm(..., ...)` with `areIdentical(..., ...)` too, choose your favourite.

Reply:
Done